### ISSUE TITLE
Error: Compatible vdso's size is bigger than reserved buf

### ISSUE THREAD

**Issue by goosesima** [checkpoint-restore/criu#1916]
**CRIU logs and information:**

<details><summary>Output of `sudo criu check --all`:</summary>
<p>

```
Error: Compatible vdso's size is bigger than reserved buf
Error (criu/vdso.c:441): vdso: Compat vdso helper exited with 255
Error (criu/vdso.c:498): vdso: Failed to mmap compatible vdso with helper process
Error (criu/vdso.c:620): vdso: Failed to fill compat vdso symtable
Error (criu/kerndat.c:1594): kerndat_vdso_fill_symtable failed when initializing kerndat.
Error (criu/crtools.c:260): Could not initialize kernel features detection.
```

</p>
</details>

<details><summary>Output of `criu --version`:</summary>
<p>

```
Version: 3.17
GitID: v3.17
```

</p>
</details>

<details><summary>Output of `criu check --all`:</summary>
<p>

```
Error (criu/util.c:641): exited, status=4
Error (criu/util.c:641): exited, status=4
Warn  (criu/proc_parse.c:930): Write 4294967295 to /proc/self/loginuid failed: Operation not permitted
Warn  (criu/net.c:3427): Unable to get tun network namespace
Warn  (criu/sk-unix.c:224): unix: Unable to open a socket file: Operation not permitted
Error (criu/net.c:3711): Unable create a network namespace: Operation not permitted
Warn  (criu/net.c:3767): NSID isn't reported for network links
Error: Compatible vdso's size is bigger than reserved buf
Error (criu/vdso.c:441): vdso: Compat vdso helper exited with 255
Error (criu/vdso.c:498): vdso: Failed to mmap compatible vdso with helper process
Error (criu/vdso.c:620): vdso: Failed to fill compat vdso symtable
Error (criu/kerndat.c:1594): kerndat_vdso_fill_symtable failed when initializing kerndat.
Error (criu/crtools.c:260): Could not initialize kernel features detection.
```

</p>
</details>

<details><summary>Output of `zcat /proc/config.gz | grep CHECKPOINT`:</summary>
<p>

```
CONFIG_CHECKPOINT_RESTORE=y
```

</p>
</details>

**Additional environment details:**
OS: Arch Linux x86_64
Kernel: 5.18.3-zen1-1-zen

**Comment by avagin** [comment]
@0x7f454c46 could you take a look at this issue if you have time?

**Comment by minhbq-99** [comment]
I think the issue is probably that in your machine the compat vdso size is larger than 2 pages. CRIU sets the upper bound of compat vdso size to 2 pages so it fails when vdso size is larger. It can be easily fixed by increasing the upper bound in CRIU.

I tried to get the vdso on vagrant arch linux (Linux arch.localdomain 5.18.10-arch1-1) by running a simple C 32 bit C program and run `cat /proc/<pid>/maps`
`f7f84000-f7f86000 r-xp 00000000 00:00 0                                  [vdso]`
The vdso size is 2 pages in size in my machine.

If it's possible, can you do the following stuffs the get the vdso size in your machine

The C program
```c
#include <stdio.h>

int main()
{
	puts("Press ENTER to continue");
	scanf("%*c");
	return 0;
}
```
Then compile it with `gcc -m32 -o test test.c`, run `test` and in another terminal read the `/proc/<pid>/maps` where <pid> is the pid of `test`.

**Comment by goosesima** [comment]
@[minhbq-99](https://github.com/minhbq-99)
```
565c6000-565c7000 r--p 00000000 103:02 1498758                           /test
565c7000-565c8000 r-xp 00001000 103:02 1498758                           /test
565c8000-565c9000 r--p 00002000 103:02 1498758                           /test
565c9000-565ca000 r--p 00002000 103:02 1498758                           /test
565ca000-565cb000 rw-p 00003000 103:02 1498758                           /test
57588000-575aa000 rw-p 00000000 00:00 0                                  [heap]
f7c00000-f7c20000 r--p 00000000 103:02 1051736                           /usr/lib32/libc.so.6
f7c20000-f7da5000 r-xp 00020000 103:02 1051736                           /usr/lib32/libc.so.6
f7da5000-f7e23000 r--p 001a5000 103:02 1051736                           /usr/lib32/libc.so.6
f7e23000-f7e24000 ---p 00223000 103:02 1051736                           /usr/lib32/libc.so.6
f7e24000-f7e26000 r--p 00223000 103:02 1051736                           /usr/lib32/libc.so.6
f7e26000-f7e27000 rw-p 00225000 103:02 1051736                           /usr/lib32/libc.so.6
f7e27000-f7e31000 rw-p 00000000 00:00 0
f7fa0000-f7fa2000 rw-p 00000000 00:00 0
f7fa2000-f7fa6000 r--p 00000000 00:00 0                                  [vvar]
f7fa6000-f7fa9000 r-xp 00000000 00:00 0                                  [vdso]
f7fa9000-f7fab000 r--p 00000000 103:02 1050742                           /usr/lib32/ld-linux.so.2
f7fab000-f7fd0000 r-xp 00002000 103:02 1050742                           /usr/lib32/ld-linux.so.2
f7fd0000-f7fdf000 r--p 00027000 103:02 1050742                           /usr/lib32/ld-linux.so.2
f7fdf000-f7fe1000 r--p 00035000 103:02 1050742                           /usr/lib32/ld-linux.so.2
f7fe1000-f7fe2000 rw-p 00037000 103:02 1050742                           /usr/lib32/ld-linux.so.2
ffe42000-ffe64000 rw-p 00000000 00:00 0                                  [stack]
```

**Comment by minhbq-99** [comment]
Thank you. Your vdso size is 3 pages which causes the error. I created a PR to fix the error.
